server/webapp/WEB-INF/rails.new/app/assets/javascripts/pipeline_configs/models/materials.js

```diff
@@ -107,6 +105,52 @@ define(['mithril', 'lodash', 'string-plus', './model_mixins', './encrypted_value
       throw new Error("Subclass responsibility!");
     };
 
+    this._passwordHash = function() {
+      if (this.isPlainPasswordValue() || this.isDirtyPasswordValue()) {
+        return { password: this.passwordValue() };
+      }
+      return { encryptedPassword: this.passwordValue() };
+    };
+
+    this.testConnection = function (vm, pipelineName) {
```

Comment:
This must be pulled out of here as 
* model should not be handing anything related to the view
* this need not be an instance method. Rather can take in all the parameters and return a promise (if done via m.request)